service/service_construction.go

```diff
@@ -412,14 +424,35 @@ func (s ConstructionService) ConstructionPayloads(
 		return nil, wrapError(errInvalidInput, fmt.Errorf("%s is not a valid address", toAddress))
 	}
 
-	tx := ethtypes.NewTransaction(
-		nonce,
-		ethcommon.HexToAddress(checkTo),
-		amount,
-		gasLimit,
-		gasPrice,
-		transferData,
-	)
+	var tx *ethtypes.Transaction
```

Comment:
The assignment of `tx` could be cleaned up. The if statement should assign `transferData` and the `ethcommon.HexToAddress` call with a single assignment of `tx` after line 455.